### PR TITLE
Make Pooled LDAP connections return to pool when closed

### PR DESCRIPTION
**Pull request by bdemers**
Fixes: DIRAPI-398

### CODE REVIEW

File: ldap/client/api/src/main/java/org/apache/directory/ldap/client/api/LdapConnectionPool.java [hunk 1]
```diff
@@ -123,6 +123,10 @@ public LdapConnection getConnection() throws LdapException
         try
         {
             connection = super.borrowObject();
+            if ( connection instanceof PooledLdapConnection )
+            {
+                ( ( PooledLdapConnection ) connection ).setConnectionPool( this );
+            }
```
Comment (bdemers) [inline ldap/client/api/src/main/java/org/apache/directory/ldap/client/api/LdapConnectionPool.java:129]: I ran into a chicken and egg problem with the pooled connection, so this was the work around...

Ideally I'd want to pass the pool in the constructor of the `LdapConnectionFactory`, but that itself is passed indirectly into the `AbstractPoolableLdapConnectionFactory`

An alternative, _might_ be to have the `LdapConnectionPool`'s constructor inspect the type `PooledObjectFactory`, if it's a `AbstractPoolableLdapConnectionFactory`, the connection factory could be wrapped 🤔
Similar solution, different location 🤷

**Comment by elecharny** [comment]
@bdemers give me a bir of time to review the PR, I'll be busy the next two days. 

Those classes are not frequently used, but every thing that makes it easier o use is good to have.

**Comment by bdemers** [comment]
No rush!  I just wanted to get the initial idea out of my head while I was thinking about it 😆

**Comment by elecharny** [comment]
Pushed to trunk. Thanks Brian!

**Comment by elecharny** [comment]
Pushed to trunk. Thanks Brian!

**Pull request closed without merging**